data_structures/reverse_a_linked_list.cpp

```diff
@@ -0,0 +1,181 @@
+/**
+ * @file
+ * @brief Implementation of  [Reversing
+ * a single linked list](https://simple.wikipedia.org/wiki/Linked_list)
+ * @details
+ * The linked list is a data structure used for holding a sequence of
+ * values, which can be added, displayed,reversed, or removed.
+ * ### Algorithm
+ * Values can be added by iterating to the end of a list (by following
+ * the pointers) starting from the first link. Whichever link points to null
+ * is considered the last link and is pointed to the new value.
+ *
+ * Linked List can be reversed by using 3 pointers: current, previous, and
+ * next_node; we keep iterating until the last node. Meanwhile, before changing
+ * to the next of current, we store it in the next_node pointer, now we store
+ * the prev pointer in the current of next, this is where the actual reversal
+ * happens. And then we move the prev and current pointers one step forward.
+ * Then the head node is made to point to the last node (prev pointer) after
+ * completion of an iteration.
+ * Graphic
+ * Explanation:https://drive.google.com/file/d/1pM5COF0wx-wermnNy_svtyZquaCUP2xS/view?usp=sharing
```

Comment:
What is this? Is this the explanation of the algorithm? 🤔

Reply:
Not an explanation exactly but a graphical view of what is happening behind the scenes.